File: abc/abc244/c.cpp

#include <bits/stdc++.h>

using namespace std;

int main(){
  int n;
  cin >> n;

  vector<bool> ans(2*n+1, false);
  for(int i = 0; i <= n; i++) {
      for (int j=1; j <= 2*n+1; j++) {
          if (ans[j] == false) {
              cout << j << endl;
              std::flush(std::cout);
              ans[j] = true;
              break;
          }
      }
      int a;
      cin >> a;
      ans[a] = true;
  }
  return 0;
}

File: abc/abc244/c.rs
use proconio::input;
use std::io::{self, Write};

fn main() {
    input! {
        n: usize,
    }

    let mut ans = vec![false; 2*n+2];
    ans[0] = true;

    for _ in 0..=n {
        for i in 1..=2*n+1 {
            if !ans[i] {
                println!("{}", i);
                ans[i] = true;
                io::stdout().flush().unwrap();
                break
            }
        }
        input! {
            aoki: usize
        }
        ans[aoki] = true;
    }
}